images.go

```diff
@@ -197,6 +203,57 @@ func (api *API) CreateImageDirectUploadURL(ctx context.Context, accountID string
 	return imageDirectUploadURLResponse.Result, nil
 }
 
+var imagesMultipartBoundary = "----CloudflareImagesGoClientBoundary"
+
+// CreateImageDirectUploadURLV2 creates an authenticated v2 direct upload url.
+//
+// API Reference: https://developers.cloudflare.com/api/operations/cloudflare-images-create-authenticated-direct-upload-url-v-2
+func (api *API) CreateImageDirectUploadURLV2(ctx context.Context, accountID string, params ImageDirectUploadURLV2Request) (ImageDirectUploadURL, error) {
```

Comment:
can we update this to align with the library conventions ([link](https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md))?

i don't love this method name but due to the version existing in the URL, we don't really have much else to offer here. i'll chat with the team internally and see what we can do here.

Reply:
...maybe a `Version` field on the params struct that toggles the URL and we roll this into the existing create method?

Reply:
The `CreateImageDirectUploadURLV2` method is significantly different from the existing method and may result into a series of if/else blocks and extracted private methods specific to each version. The request payload is also different between the two versions. The current proposed changes seems like the most clean way to do this. I'm open to other suggestions?

I've updated `ImageDirectUploadURLV2Request` with library conventions to use pointers for `bool` and `time.Time`. Was there anything else you wanted to change to meet the conventions?

Reply:
we don't want V1/V2/V3 as part of the method call since that should be abstracted from end users and they should just be able to configure/control the method as needed instead. i'm fine with private methods that we invoke from inside the shared method with a specific comment documenting it.

as for the library conventions, it looks like the only one missing is the second parameter should be `ResourceContainer`, not `string` accountID.

Reply:
Thank you for the input, i've made necessary changes. I ended up keeping all the implementation in the same function to keep it simple for now. Also updated the second parameter to by `ResourceContainer` instead of string. Let me know if there are other changes needed.